### DISCUSSION TITLE
2.0.0 Beta

### DISCUSSION THREAD

**Discussion by timolins** [timolins/react-hot-toast#81] [General]
- Add tailwind animation for examples  cfbf32a
- Fix sticky nav in docs  03d75e9
- Use activeClass for pointer events  2036a3f
- Use children instead of render prop  8388051
- Extract default offset to variable  ff9a762
- Use flexbox for toast positioning  9381c26
- Expose `gutter` prop  273e29e
- Reduce custom toast duration to 4 seconds  466825d
- Scope aria props and add render function support  f1d367c
- Add reduce motion support Closes #34  0363011
- Merge remote-tracking branch &#39;origin/main&#39; into dev  39f85aa
- Merge pull request #72 from timurmaio/main  ad291cf
- Footer (fix year)  1b2e82a
- Use dynamic year in footer  6639eea
- Allow position to be configured per toast  80e936f
- Merge branch &#39;main&#39; into dev  1a584db
- Fix invalid types/imports  664bcbf
- Refactor `Indicator` to `ToastIcon` component - Also export resolveValue to render custom JSX content  25690e0
- Add support for `toast.custom` - Render any JSX without additional styles  b7509bd
- Use React.CSSProperties instead csstype  e58d6a8
- Add `containerClassName` prop - Closes #43  2be3b23
- Add support for per toast position - Breaking: Remove visibleToasts - Rename `margin` to `gutter`  345aac0
- Add `renderToast` option to Toatser - Allows you to replace default toast with custom component (Closes #13)  207bf66
- Set `sideEffects: false` in package.json  641369f
- Breaking: No longer expose `dispatch`  8f82fec
- Update goober &amp; add `csstype` as dev dependency Closes #44  d60adb6
- Improve animations and exit positioning - Hidden toasts now stay at their position - Use percentages instead of px for animation  396db8b
- Merge branch &#39;lift-position-wrapper&#39; into dev  41aac06
- Change toast positioning to absolute  781b16f
- Merge pull request #53 from timolins/clear-timeouts  351e62a
- Move height ref to &lt;Toaster/&gt;  06e271c
- Update use-toaster.mdx  1ed8c8e
- Merge branch &#39;lift-position-wrapper&#39; into dev  2d52e4e
- Merge branch &#39;clear-timeouts&#39; into dev  8a3ac8c
- Merge branch &#39;infinity-loading&#39; into dev  3836611
- Merge pull request #49 from gr-qft/patch-1  2b8f7c1
- Increase default duration from loading toasts to Infinity Fixes #45  633cb6f
- Keep track of dismissed toasts  176bc05
- Merge pull request #52 from tmcw/patch-1  1609cc6
- Fix typo in Toaster docs  cf1fd4c
- Fix a typo  e1c68b6
- Merge pull request #48 from domenicrosati/patch-1  bfa5129
- fix spelling error  619b508
- Merge pull request #37 from timolins/persistent-toast  3e2c069
- Merge pull request #35 from hudochenkov/fix-id  8cd1535
- Update footer link  d30d2e4
- Use `Infinity` instead of `-1` for persistant toasts  a214271
- Remove dismiss from example  597a17d
- Allow `-1` duration for persistent toasts  64b5568
- Fix `undefined` id  e40ebbd
- Merge pull request #32 from Kilian/patch-2  67bdc5b
- Positioning wrapper div gets no pointer events  b441274
- Merge pull request #30 from Kilian/patch-1  854c4e2
- Move comment in toaster.mdx  b0a1da4
- add missing className to example with all available options  4239b66
- Merge remote-tracking branch &#39;origin/main&#39; into lift-position-wrapper  802fe9e
- Add missing goober setup  b43caa1
- Lift position wrapper to Toaster  aa3c0c6

https://github.com/timolins/react-hot-toast/compare/v1.0.2...v2.0.0-beta.0

<hr /><em>This discussion was created from the release <a href='https://github.com/timolins/react-hot-toast/releases/tag/v2.0.0-beta.0'>2.0.0 Beta</a>.</em>

**Comment by rishimohan** [comment]
The ability to position each toaster in 2.0 is awesome, was looking for it.

Any plans on being able to assign `onClick`, `onShow` actions to `toast()`, that would be really helpful in most cases.

Also appreciate the work you've done, it's the smoothest toaster library ever.
